src/pages/index.js

```diff
@@ -93,22 +193,77 @@ const editProfilePopup = new PopupWithForm({
 });
 editProfilePopup.setEventListeners();
 
+profileEditBtn.addEventListener("click", () => {
+  const userData = userInfo.getUserInfo();
+  profileInputTitle.value = userData.title;
+  profileInputDescription.value = userData.description;
+  profileEditFormValidator.resetValidation();
+  editProfilePopup.open();
+});
+
+// AVATAR MODAL AND METHODS -----
+const handleAvatarFormSubmit = (data) => {
+  editAvatarPopup.renderLoading(true);
+  api
+    .updateProfileAvatar(data.link)
+    .then(() => {
+      userInfo.changeAvatar(data.link);
+      editAvatarPopup.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      addCardPopup.renderLoading(false);
+    });
+};
+
+const editAvatarPopup = new PopupWithForm({
+  popupSelector: "#edit-avatar-modal",
+  handleFormSubmit: handleAvatarFormSubmit,
+});
+editAvatarPopup.setEventListeners();
+
+editAvatarButton.addEventListener("click", () => {
+  avatarUrlInput.value = userInfo.getUserInfo().avatar;
+  avatarFormValidator.resetValidation();
+  editAvatarPopup.open();
+});
+
+// ADD CARD MODAL AND METHODS -----
+
 const addCardPopup = new PopupWithForm({
   popupSelector: "#card-add-modal",
   handleFormSubmit: handleAddCardFormSubmit,
 });
 addCardPopup.setEventListeners();
 
+function handleAddCardFormSubmit(data) {
+  addCardPopup.renderLoading(true);
+  api
+    .addCard({ name: data.name, link: data.link })
+    .then((cardData) => {
+      section.addItem(createCard(cardData));
+      cardAddForm.reset();
+      addCardPopup.close();
+      addCardFormValidator.resetValidation();
+    })
+    .catch(console.error)
+    .finally(() => {
+      addCardPopup.renderLoading(false);
+    });
+}
+
 cardAddNewBtn.addEventListener("click", () => {
   addCardPopup.open();
   addCardFormValidator.toggleButtonState();
-  console.log("hello, world");
 });
 
-profileEditBtn.addEventListener("click", () => {
-  const userData = userInfo.getUserInfo();
-  profileInputTitle.value = userData.name;
-  profileInputDescription.value = userData.description;
-  profileEditFormValidator.resetValidation();
-  editProfilePopup.open();
-});
+// FORM VALIDATION -----
+
+const profileEditFormValidator = new FormValidator(settings, profileForm);
```

Comment:
I tried this but the name of the form was coming back undefined. Any advice would be appreciated.

Reply:
> I tried this but the name of the form was coming back undefined. Any advice would be appreciated.

Please, add `name` attributes to each `form` tag in your `index.html`, then it will work